### PR TITLE
Feature/1082 failure when comparing nested objects

### PR DESCRIPTION
**Pull request by lwasylow**
Fixing issue where code was failing on the nested objects due to "object" type being pulled into dynamic SQL as datatype to extract into. 

Fixing issue when null value was passed to get_hash function causing compare to fail 

Address a possible scenario when more than 2 levels of nesting for same objects caused ambiguity of the column name.

Closing #1082 
Closing #1098 
Closing #1083

### CODE REVIEW

**Review by simasch — APPROVED** [review]

**Comment by jgebal** [comment]
Let me have a deeper look into that PR before I approve it

File: test/ut3_user/expectations/test_expectation_anydata.pks [hunk 1]
```diff
@@ -203,5 +203,29 @@ create or replace package test_expectation_anydata is
  
   --%test ( Empty Array not equal array with space )
   procedure arr_empty_nqua_arr_e_unord;  
+  
+  --%test ( Failure of comparing nested objects )
```
Comment (jgebal) [inline test/ut3_user/expectations/test_expectation_anydata.pks:207]: Can we name those tests better?
Maybe something like:
```
--%test(Correctly reports a diff on failure with nested object)
--%test(Correctly reports a diff on failure with double nested object)
--%test(Reports success when comparing identical nested object)
--%test(Reports success when comparing identical double nested object)
--%test(Correctly reports a diff on failure with nested object list)
```

File: test/ut3_user/expectations/test_expectation_anydata.pks [hunk 2]
```diff
@@ -203,5 +203,29 @@ create or replace package test_expectation_anydata is
...
+  procedure failure_nested_object_list;
+ 
+  --%test ( Success of comparing nested object list )
+  procedure success_nested_object_list;
+  
+  --%test(Nested VARRAYS with same data)
+  procedure nested_varray_same_data;
+  
+  --%test(Nested VARRAYS with different data)
+  procedure nested_varray_diff_data;
```
Comment (jgebal) [inline test/ut3_user/expectations/test_expectation_anydata.pks:229]: Can you add a test to cover for regression of #1098 (on nested element being null)?

**Review by jgebal — CHANGES_REQUESTED** [review]

**Comment by jgebal** [comment]
I've reviewed the PR. It addresses all 3 issues.
Brilliant work @lwasylow :ok_hand: 👍

File: test/ut3_user/expectations/test_expectation_anydata.pks (hunk 1 above)
Comment (lwasylow) [reply test/ut3_user/expectations/test_expectation_anydata.pks:207]: Good point.

File: test/ut3_user/expectations/test_expectation_anydata.pks (hunk 2 above)
Comment (lwasylow) [reply test/ut3_user/expectations/test_expectation_anydata.pks]: Thats test replicated using our types and regression created.

**Review by jgebal — APPROVED** [review]
